src/frontend/src/components/deviceListItem.ts

```diff
@@ -0,0 +1,84 @@
+import { TemplateResult, html } from "lit-html";
+import { Setting } from "../flows/manage/deviceSettings";
+import { warningIcon, dropdownIcon, lockIcon } from "./icons";
+
+// A simple representation of "device"s used on the manage page.
+export type Device = {
+  // All the settings allowed for a particular device
+  settings: Setting[];
+  // The displayed name of a device (not exactly the "alias") because
+  // recovery devices handle aliases differently.
+  label: string;
+  isRecovery: boolean;
+  isProtected: boolean;
+  warn?: TemplateResult;
+};
+
+// A device with extra information about whether another device (earlier in the list)
+// has the same name.
+export type DedupDevice = Device & { dupCount?: number };
```

Comment:
there's something odd here, shouldn't this live closer to `dedupLabels`? Feels weird to have the type in one module and the function in another one